easybuild/framework/easyconfig/tools.py

```diff
@@ -255,18 +255,32 @@ def get_paths_for(subdir=EASYCONFIGS_PKG_SUBDIR, robot_path=None):
     path_list.extend(sys.path)
 
     # figure out installation prefix, e.g. distutils install path for easyconfigs
-    eb_path = which('eb')
+
+    # prefer using path specified in $EB_SCRIPT_PATH (if defined), which is set by 'eb' wrapper script
+    eb_path = os.getenv('EB_SCRIPT_PATH')
+    if eb_path is None:
+        # try to determine location of 'eb' script via $PATH, as fallback mechanism
+        eb_path = which('eb')
+
     if eb_path is None:
         warning_msg = "'eb' not found in $PATH, failed to determine installation prefix!"
         _log.warning(warning_msg)
         print_warning(warning_msg)
     else:
-        # real location to 'eb' should be <install_prefix>/bin/eb
-        eb_path = resolve_path(eb_path)
+        # eb_path is location to 'eb' wrapper script, e.g. <install_prefix>/bin/eb
+        # so installation prefix is usually two levels up
         install_prefix = os.path.dirname(os.path.dirname(eb_path))
         path_list.append(install_prefix)
         _log.debug("Also considering installation prefix %s..." % install_prefix)
 
+        # also consider fully resolved location to 'eb' wrapper
+        # see https://github.com/easybuilders/easybuild-framework/pull/2248
+        resolved_eb_path = resolve_path(eb_path)
```

Comment:
I would argue that the resolved path should only be used if the unresolved path doesn't contain easybuild/subdir.

Sysadmins usually frown upon code that lets users see paths that they have "hidden" with symlinks for no good reason.

Code gets somewhat more complex but would in my opinion produce a cleaner result.

As is it looks now it would for --search show both our /hpc2n/eb/software/Core/EasyBuild/... path and the resolved /cvmfs/ebsw.hpc2n.umu.se/amd64_ubutnu1604_bdw/software/.... path (unless i totally misunderstood this code)

Reply:
Yeah, good point I guess, this should really be a fallback mechanism used "in times of need".

See changes in 1a7c4fd26